docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,357 @@
+inside, you can either:
+
+1. first add a new helper: `29f49cbe9dfa <https://github.com/scality/metalk8s/commit/29f49cbe9dfa0b824c818d25d4a2f6965351e65d>`_
+
+2. then use it in new code: `7e47310a8f20 <https://github.com/scality/metalk8s/commit/7e47310a8f20fd49f0ad36707b20e6c2a53df638>`_
+
+Or:
+
+1. first add the new code: `5b2a6d5fa498 <https://github.com/scality/metalk8s/commit/5b2a6d5fa49815180a2effdd37cb58542e83b5a5>`_
+
+2. then refactor the now duplicated code: `ac08d0f53a83 <https://github.com/scality/metalk8s/commit/ac08d0f53a835a0b2bc61c1fe5b7317bf4d6550c>`_
+
+If you add tests (and you should), they can be introduced in a dedicated commit
+as well.
+
+Mixing unrelated changes
+::::::::::::::::::::::::
+
+It is sometimes tempting to do small unrelated changes as you are working on
+something else in the same code area.
+Please refrain to do so, or at least do it in a dedicated commit.
+
+Mixing non-related changes into the same commit makes revert and cherry-pick
+harder.
+
+The pull request `#1846 <https://github.com/scality/metalk8s/pull/1846>`_ is a good example. It tackles three issues at once: `#1830 <https://github.com/scality/metalk8s/issues/1830>`_
+and `#1831 <https://github.com/scality/metalk8s/issues/1831>`_ (because they are similar) and `#839 <https://github.com/scality/metalk8s/issues/839>`_ (because it was making the other
+changes easier), but it uses distincts commits for each issue.
+
+How to write a commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Why do we need commit message
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After comments in the code, commit messages are the easiest way to find context
+for every single line of code: running ``git blame`` on a file will give you,
+for each line, the identifier of the last commit that changed the line.
+
+Unlike a comment in the code (which applies to a single line or file), a commit
+message applies to a logical change and thus can provide information on the
+design of the code and why the change was done. This makes commit messages a
+part of the code documentation and makes them helpful for other developers to
+understand your code.
+
+Last but not least: commit messages can also be used for automating tasks such
+as issue management and changelog generation.
+
+Note that it is important to have all the necessary information in the commit
+message, instead of having them (only) in the related issue, because:
+
+- issues can contains troubleshooting/design discussion/investigation with a
+  lot of back and forth, which makes hard to get the gist of it.
+
+- you need access to an external service to get the whole context, which goes
+  against one of biggest advantage of the distributed SCM (having all the
+  information you need offline, from your local copy of the repository).
+
+Anatomy of a good commit message
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A commit is composed of a subject, a body and a footer. A blank line separates
+the subject from body.
```

Comment:
And what separates the footer from the body?

Reply:
A blank line as well, I'll add it.